pipeline_dp/accumulator.py

```diff
@@ -174,6 +185,47 @@ def compute_metrics(self) -> float:
 
 
 @dataclass
+class VectorSummationParams:
+    noise_kind: NoiseKind
+    max_norm: float
+
+
+class VectorSummationAccumulator(Accumulator):
+    _vec_sum: np.ndarray
+
+    def __init__(self, params: VectorSummationParams,
+                 values: Iterable[Tuple[float]]) -> None:
+        self.sum_params = params
+        values = iter(values)
+        try:
+            self._vec_sum = np.array(next(values))
+            for val in values:
+                self.add_value(val)
+        except StopIteration:
+            self._vec_sum = None
+
+    def add_value(self, value: Tuple[float]):
+        value = np.array(value)  # type: np.ndarray
```

Comment:
I'm thinking, whether np.array is better (less space, more effective?) than tuple for serializing and sending between workers, don't you know?

Reply:
`np.ndarray` is hands down better. Less memory, enjoys low level and hardware vectorization. I'm not certain if all serializations on it are as compact, but at the very least:
```python
import numpy as np
import pickle

# probably around 8000 byes of memory including metadata like dtype, shape and strides
x = np.arange(1000, dtype=np.int64)

x_p = pickle.dumps(x)
print(len(x_p)) # 8152

x_t = tuple(x) # same array but in tuple
x_t_p = pickle.dumps(x_t)
print(len(x_p)) # 19100
```
My assumption is that numpy is extremely optimized for computation and data movement, whether it's with or without multiprocessing.

Reply:
Wow. Thanks for checking! Yeah, np arrays look much better